### PR TITLE
Closes #11339: don’t stop private download when the tab is closed

### LINKED ISSUE #11339: [Bug]: Download Stops after closing its Tab in Private Window
The issue mirrors the [...fenix/issue/22395](https://github.com/mozilla-mobile/fenix/issues/22395).
Basically, closing a private tab that started a download process results in cancelling the download.
The expected behavior would be to keep downloads running until the last private tab is closed (notifying the user that closing the tab will also cancel the downloads which will be implemented [...fenix/issue/18160](https://github.com/mozilla-mobile/fenix/issues/18160)

### PR DESCRIPTION
**Pull request by mavduevskiy**
Relates to [...android-components/issues/11339](https://github.com/mozilla-mobile/android-components/issues/11339). 

Currently we stop the download when the tab is closed. The users, however, expect downloads to be active until the last private tab is open.

### CODE REVIEW

File: components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt [hunk 1]
```diff
@@ -75,7 +76,12 @@ class DownloadMiddleware(
             is TabListAction.RemoveAllTabsAction,
             is TabListAction.RemoveAllPrivateTabsAction -> removePrivateNotifications(context.store)
             is TabListAction.RemoveTabsAction -> removePrivateNotifications(context.store, action.tabIds)
-            is TabListAction.RemoveTabAction -> removePrivateNotifications(context.store, listOf(action.tabId))
+            is TabListAction.RemoveTabAction -> {
+                val privateTabs = context.store.state.getNormalOrPrivateTabs(private = true)
+                if (privateTabs.isEmpty()) {
+                    removePrivateNotifications(context.store)
+                }
+            }
```
Comment (Amejia481) [inline components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt:84]: This looks good,  I think we should add tests to cover this block :).

**Review by Amejia481 — COMMENTED** [review]
I think we also need to add some UI for showing a dialog when it's the last tab and the user is trying to close it.

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -1,179 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-// If you ever need to force a toolchain rebuild (taskcluster) then edit the following comment.
-// FORCE REBUILD 2021-10-06
-
-// Synchronized version numbers for dependencies used by (some) modules
-object Versions {
```
Comment (csadilek) [inline buildSrc/src/main/java/Dependencies.kt:9]: You're removing this file by mistake I assume? :)

Comment (mavduevskiy) [reply buildSrc/src/main/java/Dependencies.kt]: Oh, yes! Thanks for pointing that out.

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**